backend/src/main/java/planit/people/preparation/GoogleConnector/GoogleHelper.java

```diff
@@ -40,34 +42,110 @@ public String getRefreshToken() {
+        Date startTime = FreeTimeFinder.getBestStartDate(schedulingInfos, durationInMinutes);
+        assert startTime != null;
+        return new DateTime(startTime);
     }
 
-    private List<Interval> getFreeBusyForAll(Date startDate, Date endDate, Set<String> refreshTokens) throws IOException {
-        List<Interval> busyForAll = new ArrayList<>();
-        for (String refreshToken : refreshTokens) {
-            busyForAll.addAll(getFreeBusy(startDate, endDate, refreshToken));
+    /**
+     * Get free busy for all registered guests asynchronously using CompletableFuture. 
+     * The method iterate through all refresh token for each guest and get all FreeBusy responses for each Refresh Token asynchronously.
+     * 
+     * @param startDate the date after which an event should be scheduled 
+     * @param endDate the date before which an event should be scheduled. 
+     * @param usersRefreshToken a map containing the obligation to attend for all registered guests and the refresh token for all of their google accounts. The map structure is the following: Map<Obligation to attend, Map<PlanIt User Id, Set<Refresh Token>>>
+     * @return Map<String, Map<Long, List<Interval>>> a map containing all FreeBusy intervals mapped to their PlanIt User Id, which mapped to the obligation of attending the event. Map<Obligation to attend, Map<PlanIt User Id, List<FreeBusy Intervals>>>
+     * @see GoogleHelper#getFreeBusy
+     * @see Converter#convertFutureMapToIntervalMap
+     * @see CompletableFuture
+     */
+    private Map<String, Map<Long, List<Interval>>> getFreeBusyIntervalForAll(Date startDate, Date endDate, Map<String, Map<Long, Set<String>>> usersRefreshToken) throws ExecutionException, InterruptedException {
+        Map<String, Map<Long, List<CompletableFuture<List<Interval>>>>> futureCallsMap = new HashMap<>();
+        List<CompletableFuture<List<Interval>>> futureCalls = new ArrayList<>();
+        int size = -1;
+        for (String obligation : usersRefreshToken.keySet()) {
+            futureCallsMap.put(obligation, new HashMap<>());
+            for (Long userId : usersRefreshToken.get(obligation).keySet()) {
+                futureCallsMap.get(obligation).put(userId, new ArrayList<>());
+                List<CompletableFuture<List<Interval>>> futureIntervalForAUser = new ArrayList<>();
+                for (String refreshToken : usersRefreshToken.get(obligation).get(userId).stream().toList()) {
+                    size += 1;
+                    CompletableFuture<List<Interval>> futureCall = CompletableFuture.supplyAsync(() -> {
+                        try {
+                            return getFreeBusy(startDate, endDate, refreshToken);
+                        } catch (IOException e) {
+                            throw new RuntimeException(e);
+                        }
+                    });
+                    futureCalls.add(futureCall);
```

Comment:
Why do you keep `size` variable. As far as I understand each time `size` is incremented you also add elements to `futureCalls` and `futureIntervalForAUser` so size can be infered from that